enclave-runtime/src/rpc/author/mod.rs

```diff
@@ -97,76 +100,69 @@ pub trait OnBlockCreated {
 	fn on_block_created(&self, hashes: &[Self::Hash], block_hash: SidechainBlockHash);
 }
 
-/// Authoring API
-pub struct Author<TopPoolGetter>
+/// Currently we treat all RPC operations as externals.
+///
+/// Possibly in the future we could allow opt-in for special treatment
+/// of such operations, so that the block authors can inject
+/// some unique operations via RPC and have them included in the pool.
+const TX_SOURCE: TrustedOperationSource = TrustedOperationSource::External;
+
+/// Authoring API for RPC calls
+///
+///
+pub struct Author<TopPool, StateFacade, EncryptionKey>
 where
-	TopPoolGetter: GetTopPool + Sync + Send + 'static,
+	TopPool: TrustedOperationPool + Sync + Send + 'static,
+	StateFacade: HandleState,
+	EncryptionKey: ShieldingCrypto,
 {
-	top_pool_getter: Arc<TopPoolGetter>,
+	top_pool: Arc<TopPool>,
+	state_facade: Arc<StateFacade>,
+	encryption_key: EncryptionKey,
 }
```

Comment:
All of the author's dependencies are now captured as members, no implicit static/global dependencies anymore.